File: ledger-service/src/main/java/zachkingcade/dev/ledger/adapter/in/web/dto/pendingtransaction/apply/ApplyPendingTransactionItemRequest.java
package zachkingcade.dev.ledger.adapter.in.web.dto.pendingtransaction.apply;

import zachkingcade.dev.ledger.adapter.in.web.dto.journal.JournalLineDTORequest;

import java.util.List;
import java.util.Optional;

public record ApplyPendingTransactionItemRequest(
        Long pendingTransactionNumber,
        String entryDate,
        String description,
        Optional<String> notes,
        List<JournalLineDTORequest> journalLines
) {}

File: user-interface-service/src/app/adapter/ledger-service/dto/pending-transaction/apply/ApplyPendingTransactionItemRequest.ts
import { JournalLineDTORequest } from '../../journal-entry/JournalLineDTORequest';

export interface ApplyPendingTransactionItemRequest {
  pendingTransactionNumber: number;
  entryDate: string;
  description: string;
  notes?: string;
  journalLines: JournalLineDTORequest[];
}

